src/wast-parser.cc

```diff
@@ -934,7 +934,7 @@ Result WastParser::ParseDataModuleField(Module* module) {
     CHECK_RESULT(ParseOffsetExpr(&field->data_segment.offset));
   } else if (!ParseOffsetExprOpt(&field->data_segment.offset)) {
     if (options_->features.bulk_memory_enabled()) {
-      field->data_segment.flags |= SegPassive;
+      field->data_segment.kind = SegmentKind::Passive;
```

Comment:
How about flipping the condition which would result in an early return and removing the else block?

Reply:
Done.